From c8d0d3fa946976c6bc69589375d7d063f0cb1492 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <nix.or.die@googlemail.com>
Date: Wed, 23 Dec 2020 01:53:15 +0100
Subject: [PATCH] hwmon: (k10temp) Zen3 Ryzen Desktop CPUs support
Git-commit: c8d0d3fa946976c6bc69589375d7d063f0cb1492
Patch-mainline: v5.12-rc2
References: bsn#12
Modified-by-SEL: No


The module has only support for Zen3 server CPUs right now.
Add support for Family 0x19, model 0x21 which are Zen3 Ryzen Desktop CPUs.
Tested on 5800x, 5900x and 5950x CPUs.

Signed-off-by: Gabriel Craciunescu <nix.or.die@gmail.com>
Link: https://lore.kernel.org/r/20201223005315.20077-1-nix.or.die@gmail.com
Signed-off-by: Guenter Roeck <linux@roeck-us.net>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/hwmon/k10temp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/hwmon/k10temp.c b/drivers/hwmon/k10temp.c
index 3bc2551577a3..5ff3669c2b60 100644
--- a/drivers/hwmon/k10temp.c
+++ b/drivers/hwmon/k10temp.c
@@ -448,7 +448,8 @@ static int k10temp_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		data->is_zen = true;
 
 		switch (boot_cpu_data.x86_model) {
-		case 0x0 ... 0x1:	/* Zen3 */
+		case 0x0 ... 0x1:	/* Zen3 SP3/TR */
+		case 0x21:		/* Zen3 Ryzen Desktop */
 			k10temp_get_ccd_support(pdev, data, 8);
 			break;
 		}
-- 
2.33.0

